c7n/resources/awslambda.py

```diff
@@ -278,6 +283,42 @@ class KmsFilter(KmsRelatedFilter):
     RelatedIdsExpression = 'KMSKeyArn'
 
 
+@AWSLambda.action_registry.register('post-finding')
+class LambdaPostFinding(PostFinding):
+
+    resource_type = 'AwsLambdaFunction'
+
+    def format_resource(self, r):
+        envelope, payload = self.format_envelope(r)
+        # security hub formatting beggars belief
+        details = self.filter_empty(select_keys(r,
+            ['CodeSha256',
+             'DeadLetterConfig',
+             'Environment',
+             'Handler',
+             'KMSKeyArn',
+             'LastModified',
+             'MemorySize'
+             'MasterArn',
+             'RevisionId',
+             'Role',
+             'Runtime',
+             'TracingConfig',
+             'Timeout',
+             'Version',
+             'VpcConfig']))
+        # do the brain dead parts Layers, Code, TracingConfig
+        if 'Layers' in r:
+            r['Layers'] = {
+                'Arn': r['Layers'][0]['Arn'],
+                'CodeSize': r['Layers'][0]['CodeSize']}
+        details.get('VpcConfig', {}).pop('VpcId', None)
+        if 'Code' in r:
+            details['Code'] = {'ZipFile': r['Code']['Location']}
```

Comment:
Out of curiosity, `r['Code']['Location']` the base64-encoded contents of the deployment package? Or is it the bucket location?

Reply:
thats nuts re base64 content, thats could be hundreds of mb. they haven't thought that through at all imo. @FireballDWF ^

for any direct upload/console functions, there is no s3 bucket/location per se, I guess we have to url parse and do a bunch of assumptions to extract a bucket, key and version.. but the key won't be accessible to them, cause its a shared bucket...

Reply:
like much of the sec hub arbitrary attribute mapping, parsing this to s3 locations/keys/versions is pretty useless because its a shared bucket from a service account that can't be access except by signed url.

Reply:
Interesting. I wonder why they even collect this info in the finding. Seems like function name, account id and region would be enough to point to the offending lambda. Looks like it’s optional though and could likely be omitted.